main/test/ca/uwaterloo/flix/library/TestSet.flix

```diff
@@ -0,0 +1,222 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/////////////////////////////////////////////////////////////////////////////
+// size                                                                    //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def size01: Bool = Set/size(#{}) `assertEq!` 0
+
+@test
+def size02: Bool = Set/size(#{1}) `assertEq!` 1
+
+/////////////////////////////////////////////////////////////////////////////
+// empty                                                                   //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def empty01: Bool = Set/empty() `assertEq!` #{}
+
+/////////////////////////////////////////////////////////////////////////////
+// singleton                                                               //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def singleton01: Bool = Set/singleton(1) `assertEq!` #{1}
+
+@test
+def singleton02: Bool = Set/singleton(2) `assertEq!` #{2}
+
+@test
+def singleton03: Bool = Set/singleton(true) `assertEq!` #{true}
+
+/////////////////////////////////////////////////////////////////////////////
+// insert                                                                  //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def insert01: Bool = Set/insert(1, #{}) `assertEq!` #{1}
+
+@test
+def insert02: Bool = Set/insert(1, #{1}) `assertEq!` #{1}
+/*
+@test
+def insert03: Bool = Set/insert(1, #{2}) `assertEq!` #{1, 2}
+
+@test
+def insert04: Bool = Set/insert(1, #{2, 3}) `assertEq!` #{1, 2, 3}
+
+@test
+def insert05: Bool = Set/insert(1, #{1, 2, 3}) `assertEq!` #{1, 2, 3}
+
+@test
+def insert06: Bool = Set/insert(1, #{2, 1, 3}) `assertEq!` #{2, 1, 3}
+*/
+
+/////////////////////////////////////////////////////////////////////////////
+// delete                                                                  //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def delete01: Bool = Set/delete(1, #{}) `assertEq!` #{}
+
+@test
+def delete02: Bool = Set/delete(1, #{1}) `assertEq!` #{}
+
+@test
+def delete03: Bool = Set/delete(1, #{2}) `assertEq!` #{2}
+
+/////////////////////////////////////////////////////////////////////////////
+// null                                                                    //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def null01: Bool = Set/null(#{}) `assertEq!` true
```

Comment:
I think you should just use plain old `assert!`. The reason is used `assertEq!` with `true` and `false` in the example I gave you was because I was working with boolean functions and I wanted to be explicit. I think

```
assert!(Set/null(#{})
```

reads just fine.